tests/unit/useOnyxTest.ts

```diff
@@ -257,6 +255,8 @@ describe('useOnyx', () => {
             expect(result.current[0]).toEqual('initial value');
             expect(result.current[1].status).toEqual('loaded');
 
+            Onyx.set(ONYXKEYS.TEST_KEY, 'test_id_1');
```

Comment:
Just curious: Why did you move the operation to here? Shouldn't we have tests for both cases?

Reply:
The moment we use `Onyx.set` or `Onyx.merge` (or any setter function) it will store the value in cache synchronously and update the `storageKeys` Set in `OnyxCache`.\

If we set the key before asserting for the `initialValue`, we will always get `undefined`, because there IS a value in cache already, even though the storage promise hasn't resolved yet.

Reply:
Thanks for the explanation!